java/ql/src/semmle/code/java/frameworks/spring/SpringWebUtil.qll

```diff
@@ -0,0 +1,210 @@
+/** Provides models of taint flow in `org.springframework.web.util` */
+
+import java
+private import semmle.code.java.dataflow.ExternalFlow
+
+private class FlowSummaries extends SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      [
+        "org.springframework.web.util;UriBuilder;true;build;;;Argument[-1];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;build;(Map);;MapValue of Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;fragment;;;Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriBuilder;true;fragment;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;host;;;Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriBuilder;true;host;;;Argument[0];ReturnValue;value",
+        "org.springframework.web.util;UriBuilder;true;path;;;Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriBuilder;true;path;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;pathSegment;;;Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriBuilder;true;pathSegment;;;ArrayElement of Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;port;;;Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriBuilder;true;port;(java.lang.String);;Argument[0];Argument[-1];value",
+        "org.springframework.web.util;UriBuilder;true;query;;;Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriBuilder;true;query;;;Argument[0];Argument[-1];value",
```

Comment:
```suggestion
        "org.springframework.web.util;UriBuilder;true;query;;;Argument[0];Argument[-1];taint",
```

Reply:
Are the tests even succeeding with these errors?

Reply:
No, sorry.